website/routes.py

```diff
@@ -197,11 +202,26 @@ def ember_app(path=None):
     """Serve the contents of the ember application"""
     ember_app_folder = None
     fp = path or 'index.html'
-    for k in settings.EXTERNAL_EMBER_APPS.keys():
+
+    ember_app = None
+
+    for k in EXTERNAL_EMBER_APPS.keys():
         if request.path.strip('/').startswith(k):
-            ember_app_folder = os.path.abspath(os.path.join(os.getcwd(), settings.EXTERNAL_EMBER_APPS[k]['path']))
+            ember_app = EXTERNAL_EMBER_APPS[k]
             break
 
+    if not ember_app:
+        raise HTTPError(http.NOT_FOUND)
+
+    if settings.PROXY_EMBER_APPS:
+        url = urlparse.urljoin(ember_app['server'], request.path[len(ember_app['path']):])
+        resp = requests.get(url, stream=True, timeout=EXTERNAL_EMBER_SERVER_TIMEOUT, headers={'accept': 'text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8'})
+        excluded_headers = ['content-encoding', 'content-length', 'transfer-encoding', 'connection']
+        headers = [(name, value) for (name, value) in resp.raw.headers.items() if name.lower() not in excluded_headers]
+        return Response(resp.content, resp.status_code, headers)
```

Comment:
Passing the headers through in the response was what I was missing when I tried this. I think the browser was trying to parse the .js files as HTML, which would explain why it was so slow. Nice job @binoculars and @icereval!

Reply:
The express server in ember-cli tries to do a keepalive connection, which is at least partially responsible for it timing out.

This is what we used to have: https://github.com/CenterForOpenScience/osf.io/commit/b6ce05f1a0d875f181fcad1e6b744caef9cb1ac9#diff-758f58f82207e3674b3105d2b3d4ce7a